### ISSUE TITLE
generate-event: Add Application Load Balancer

### ISSUE THREAD

**Issue by michaelbrewer** [aws/aws-sam-cli#3671]
### Describe your idea/feature/enhancement

Application load balancer payload is missing from `sam local generate-event`

### Proposal

Add Application load balancer payload event to generate-event

### Additional Details

- [Example Application Load Balancer request event](https://docs.aws.amazon.com/lambda/latest/dg/services-alb.html)

```json5
{
    "requestContext": {
        "elb": {
            "targetGroupArn": "arn:aws:elasticloadbalancing:us-east-2:123456789012:targetgroup/lambda-279XGJDqGZ5rsrHC2Fjr/49e9d65c45c6791a"
        }
    },
    "httpMethod": "GET",
    "path": "/lambda",
    "queryStringParameters": {
        "query": "1234ABCD"
    },
    "headers": {
        "accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8",
        "accept-encoding": "gzip",
        "accept-language": "en-US,en;q=0.9",
        "connection": "keep-alive",
        "host": "lambda-alb-123578498.us-east-2.elb.amazonaws.com",
        "upgrade-insecure-requests": "1",
        "user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/71.0.3578.98 Safari/537.36",
        "x-amzn-trace-id": "Root=1-5c536348-3d683b8b04734faae651f476",
        "x-forwarded-for": "72.12.164.125",
        "x-forwarded-port": "80",
        "x-forwarded-proto": "http",
        "x-imforwards": "20"
    },
    "body": "",
    "isBase64Encoded": false
}
```

- [alb-lambda-target-request-headers-only.json](https://github.com/aws/aws-lambda-go/blob/main/events/testdata/alb-lambda-target-request-headers-only.json)

```json5
{
  "requestContext": {
    "elb": {
      "targetGroupArn": "arn:aws:elasticloadbalancing:us-east-1:123456789012:targetgroup/lambda-target/abcdefg"
    }
  },
  "httpMethod": "GET",
  "path": "/",
  "queryStringParameters": {
    "key": "hello"
  },
  "headers": {
    "accept": "*/*",
    "connection": "keep-alive",
    "host": "lambda-test-alb-1334523864.us-east-1.elb.amazonaws.com",
    "user-agent": "curl/7.54.0",
    "x-amzn-trace-id": "Root=1-5c34e93e-4dea0086f9763ac0667b115a",
    "x-forwarded-for": "25.12.198.67",
    "x-forwarded-port": "80",
    "x-forwarded-proto": "http",
    "x-imforwards": "20",
    "x-myheader": "123"
  },
  "isBase64Encoded": false
}
```

- [alb-lambda-target-request-multivalue-headers.json](https://github.com/aws/aws-lambda-go/blob/main/events/testdata/alb-lambda-target-request-multivalue-headers.json)

```json5
{
  "requestContext": {
    "elb": {
      "targetGroupArn": "arn:aws:elasticloadbalancing:us-east-1:123456789012:targetgroup/lambda-target/abcdefgh"
    }
  },
  "httpMethod": "GET",
  "path": "/",
  "multiValueQueryStringParameters": {
    "key": [
      "hello"
    ]
  },
  "multiValueHeaders": {
    "accept": [
      "*/*"
    ],
    "connection": [
      "keep-alive"
    ],
    "host": [
      "lambda-test-alb-1234567.us-east-1.elb.amazonaws.com"
    ],
    "user-agent": [
      "curl/7.54.0"
    ],
    "x-amzn-trace-id": [
      "Root=1-5c34e7d4-00ca239424b68028d4c56d68"
    ],
    "x-forwarded-for": [
      "72.21.198.67"
    ],
    "x-forwarded-port": [
      "80"
    ],
    "x-forwarded-proto": [
      "http"
    ],
    "x-imforwards": [
      "20"
    ],
    "x-myheader": [
      "123"
    ]
  },
  "body": "Some text",
  "isBase64Encoded": false
}
```

**Comment by mingkun2020** [comment]
@michaelbrewer Thanks for the contribution, we are reviewing your pr now. I think the Addition Detail section above is wrong, the above example: https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html should be for [issue#3654](https://github.com/aws/aws-sam-cli/issues/3654) and https://docs.aws.amazon.com/lambda/latest/dg/services-alb.html is for this issue.

**Comment by michaelbrewer** [comment]
Sorry @mingkun2020  i have updated it to be right. I copied it from my previous issues for http api

**Comment by benbenbenbenbenben** [comment]
Any idea when we can expect this to merge?

**Comment by hawflau** [comment]
Released in [SAM CLI v1.43.0](https://github.com/aws/aws-sam-cli/releases/tag/v1.43.0). Closing.
